Mobile Buy SDK/Mobile Buy SDK/Models/BUYAccountCredentials.h

```diff
@@ -25,40 +25,55 @@
 //
 
 #import <Foundation/Foundation.h>
+NS_ASSUME_NONNULL_BEGIN
 
-/**
- *   Intended for storing a collection of credential items representing individual values
- */
+extern NSString * const BUYAccountFirstNameKey;
+extern NSString * const BUYAccountLastNameKey;
+extern NSString * const BUYAccountEmailKey;
+extern NSString * const BUYAccountPasswordKey;
+extern NSString * const BUYAccountPasswordConfirmationKey;
 
 @class BUYAccountCredentialItem;
-@interface BUYAccountCredentials : NSObject
 
-NS_ASSUME_NONNULL_BEGIN
+/**
+ * Encapsulates user's credentials represented by BUYAccountCredentialItem
+ * objects.
+ */
+@interface BUYAccountCredentials : NSObject
 
-+ (BUYAccountCredentials *)credentialsWithItems:(NSArray<BUYAccountCredentialItem *> *)items;
-+ (BUYAccountCredentials *)credentialsWithItemKeys:(NSArray<NSString *> *)keys;
+@property (nonatomic, assign, readonly) NSUInteger count;
+@property (nonatomic, assign, readonly, getter=isValid) BOOL valid;
+@property (nonatomic, strong, readonly) NSDictionary *JSONRepresentation;
 
-@property (readonly) NSDictionary *JSONRepresentation;
-@property (nonatomic, readonly, getter=isValid) BOOL valid;
++ (BUYAccountCredentials *)credentials;
++ (BUYAccountCredentials *)credentialsWithItems:(nullable NSArray<BUYAccountCredentialItem *> *)items;
+- (instancetype)initWithItems:(nullable NSArray<BUYAccountCredentialItem *> *)items;
 
-- (BUYAccountCredentialItem *)objectForKeyedSubscript:(NSString *)key;
-- (void)setObject:(BUYAccountCredentialItem *)obj forKeyedSubscript:(NSString *)key;
+- (void)setCredentialItems:(NSArray<BUYAccountCredentialItem *> *)items;
+- (void)setCredentialItem:(BUYAccountCredentialItem *)item;
 
 @end
 
 
 /**
- *  Represents a key and KVC-validatable value
+ * Represents a single for user's credentials such as 
+ * email or password.
  */
-
 @interface BUYAccountCredentialItem : NSObject
 
-+ (instancetype)itemWithKey:(NSString *)key value:(NSString *)value;
+@property (nonatomic, assign, readonly, getter=isValid) BOOL valid;
+@property (nonatomic, strong, readonly) NSString *key;
+@property (nonatomic, strong, readonly) NSString *value;
 
-@property (nonatomic, getter=isValid) BOOL valid;
-@property (nonatomic, strong) NSString *key;
-@property (nonatomic, strong) NSString *value;
++ (instancetype)itemEmailWithValue:(NSString *)value;
```

Comment:
How about something simpler, like `-itemWithEmail:(NSString *)email`, and similar for the others?